avalanche/distributed/distributed_consistency_verification.py

```diff
@@ -0,0 +1,108 @@
+import hashlib
+import io
+
+from typing import Tuple, TYPE_CHECKING
+
+import torch
+
+from torch.utils.data import DataLoader
+
+if TYPE_CHECKING:
+    from torch import Tensor
+    from torch.nn import Module
+    from avalanche.benchmarks import DatasetScenario
+    from torch.utils.data import Dataset
+
+
+def hash_benchmark(benchmark: 'DatasetScenario', *, 
+                   hash_engine=None, num_workers=0) -> str:
```

Comment:
Yes, I think I can move those elements to the appropriate classes.

Reply:
It seems that the only avalanche-specific hash function in that file is `hash_benchmark`. Do you think it is still appropriate to move it to `CLScenario`?

Reply:
I think it's better to reuse the class `__hash__` method if possible so that child classes can safely override its behavior if needed. Also, `hash_dataset` should work only on `AvalancheDataset` since we don't really support any other dataset.

Reply:
ok, if it's different we can keep it as is. Maybe it will be more clear to me once I see how you use it for distributed training